src/colmap/image/undistortion.cc

```diff
@@ -851,17 +851,23 @@ Camera UndistortCamera(const UndistortCameraOptions& options,
 
     for (size_t y = roi_min_y; y < roi_max_y; ++y) {
       // Left border.
-      const Eigen::Vector2d point1_in_cam =
+      const std::optional<Eigen::Vector2d> point1_in_cam =
```

Comment:
This logic will likely fail for extreme fisheye lenses, where outer pixels fail to undistort. In this case, we may have to iterate over the whole image instead.